contributor-docs/adrs/adr-013-primer-experimental-location.md

```diff
@@ -0,0 +1,62 @@
+
+As we work on new components (or rewrites of old components), we often need to start with a "experiment" and not include it in the semantically versioned (semver-ed) main bundle.
+
+[ADR-006](./adr-006-drafts.md) explains our use of "drafts", here are the relevant parts:
+
+- When we are not sure about the API of a component, we put it in a different bundle: "primer/react/drafts". [ADR 010: Merge drafts status into experimental](./adr-010-drafts-are-experimental) recommends changing the name to `experimental`
+- The contract with consumers is that you are opting into a component in which we don't have 100% confidence yet, the API and structure can change.
+
+In [ADR 008: Lowering the barrier to entry](./adr-008-experimental-components) (Sep, 2022), we decided to create room to share experimental components between memex and the monolith with a new repository ([github/experimental-react-components](https://github.com/github/experimental-react-components)). Now that [memex is part of the monolith](https://github.com/github/planning-tracking/issues/1755), that [repository has been archived](https://github.com/github/platform-ux/issues/1248) (Feb, 2023).
+
+Which leaves us with: Where should experimental components maintained by Primer live?
+
+&nbsp;
+
+### Problems with publishing experimental components in @primer/react npm package:
+
+**Incompatible with semver**: Because we haven't reached the final API or structure, we need to ship breaking changes to experimental components. In the current setup, these breaking changes are part of minor or patch releases to primer/react.
+
+On the consumer side, this makes every upgrade a potentially breaking upgrade nullifying the benefits of semver.
+
+In the case of multiple teams working on the monolith, this can mean finding breaking changes for a new experimental component in a part of the app that you're not familiar with.
+
+&nbsp;
+
+## Decision
+
+1. Experimental components, as defined by the [component lifecycle](https://primer.style/contribute/component-lifecycle#experimental), should not be part of semantically versioned npm package for `@primer/react`.
+
+2. Each experimental component should be independently published and versioned with semver. Breaking changes should be tagged as major versions. Example: `@primer/react-experimental-underlinenav`.
+
+3. The source code for all experimental components should continue to live in the same repository ([primer/react](https://github.com/primer/react)).
+
+### Tradeoffs
+
+1. The responsibility of using compatible versions of experimental components and primer/react is shifted to the consumer.
+2. This would result in lots of new packages in the `@primer` scope on npm, creating some clutter.
+
+&nbsp;
+
+### Other options considered
+
+The source code for experimental components can be hosted inside the monolith in [github/github/ui/packages](https://github.com/github/github/tree/master/ui/packages).
```

Comment:
This is where things currently reside for some components (e.g. QueryBuilder). Would the plan be to migrate those out?

Reply:
I don't know enough about QueryBuilder🤔  Is this a component maintained by primer developers?

Reply:
heh that is the question. The _Web Component_ is currently in `app/components/primer/experimental`, while the React wrapper is in `app/assets/modules/react-shared`. So... partially? 😛